### DISCUSSION TITLE
当使用SQ8索引，返回的距离是原始向量的L2距离还是量化后的？

### DISCUSSION THREAD

**Discussion by SuoSiFire** [milvus-io/milvus#17585] [Q&A and General discussion] [answered]
当使用SQ8索引，返回的距离是原始向量的L2距离还是量化后的？

**Comment by yhmo** [comment, marked as answer]
是量化后的
两条完全相同的向量在IVF_FLAT里返回的L2距离是0，而在IVF_SQ8里返回的是非0

**Reply by SuoSiFire** [reply]
但是当segment数据量没达到构建索引量级的时候，这是返回的是原始向量的L2距离，当达到量级构建索引后，返回的就是量化后的距离对吧。

**Reply by yhmo** [reply]
> 但是当segment数据量没达到构建索引量级的时候，这是返回的是原始向量的L2距离，当达到量级构建索引后，返回的就是量化后的距离对吧。

没错。这里确实会有两种结果有差异的问题。对于未建索引的segment，是精确的暴搜；但对于已建好SQ8或者PQ索引的segment，是量化后的计算。在做结果集合并时，确实可能会出现一点影响，就好比左眼远视右眼近视这种不太舒服的感觉，如果两眼都是近视反倒好一些。

**Reply by SuoSiFire** [reply]
好的，了解了，这个比喻很棒👍
